CGATPipelines/Pipeline/Control.py

```diff
@@ -68,29 +70,255 @@
+        caller = f.f_locals["__file__"] # TO DO: cgatflow 
+        # Make it easier to match the name of the command executed so that
+        # the config file can be searched in case there are more than one
+        # ini files found in writeConfig():
+        caller_name = os.path.basename(os.path.normpath(caller))
+    except KeyError:
+        f = sys._getframe(2) # if e.g. call is direct for pipeline_QTL config
+        caller = f.f_globals["__file__"]
+        caller_name = os.path.basename(os.path.normpath(caller))
+        if caller_name.endswith('.py'):
+            caller_name = caller_name.replace('.py', '')
+    #else:
+    #    print('''Unable to find path to file being executed. Probably because
+    #            CGATPipelines and the pipeline that is being executed
+    #            cannot figure out where each other lives. Raise an issue in
+    #            GitHub if possible. Exiting.''')
+    #    sys.exit()
+        # CGATPipelines have a pipe_XX/pipe_XX hierarchy, but a simplified
+        # version would only have pipe_XX/
+    pipeline_path = os.path.splitext(caller)[0]
+    pipeline_path_2 = os.path.dirname(pipeline_path)
+    # CGATPipelines have a "configuration" folder
+    # adding a glob to have a bit more flexibility
+    # TO DO: add max depth to glob recursion:
+    general_path = glob.glob(str(os.path.abspath(pipeline_path_2) +
+                                  '/**/configuration*'), recursive = True)
+
+    if not general_path:
+        general_path = [os.path.join(os.path.dirname(pipeline_path),
+            "configuration")]
+
+    # Add paths to search list:
+    config_paths.extend([pipeline_path, pipeline_path_2])
+    # Extend separately in case general_path returns more than one file:
+    config_paths.extend(general_path)
+    return(config_paths, caller_name)
+
+def writeConfigFiles(config_paths, caller_name):
     '''create default configuration files in `path`.
     '''
+    # TO DO: I've modified this function with workarounds to make it more
+    # flexible in order to find an ini file, find a configuration dir and 
+    # copy pre-run sphinx-quickstart files if they exist.
+    # Other than creating a 'report' dir, it should not change the way it is
+    # run from CGATPipelines.
+    # See also getConfigPaths() above, these run when calling the 'config' option 
+    # Antonio
+    report_dir = 'pipeline_report'
+    config_files = []
+
+    try:
+        os.mkdir(report_dir) # Sphinx config files will be copied here
+                             # CGATReport only needs its conf.py to generate the rest
+                             # though
+    except FileExistsError:
+        E.warn("directory `%s` already exists" % report_dir)
+        raise
+
+    # Look for ini file:
+    f_count = 0
+    for path in config_paths:
+        if os.path.exists(path) and os.path.isdir(path):
+            for f in os.listdir(os.path.abspath(path)):
+                if fnmatch.fnmatch(f, 'pipeline*ini'):
```

Comment:
This will pick up two pipeline.ini files e.g.:

    https://github.com/CGATOxford/CGATPipelines/blob/master/CGATPipelines/pipeline_readqc/pipeline.ini
    https://github.com/CGATOxford/CGATPipelines/blob/master/CGATPipelines/configuration/pipeline.ini

The specific pipeline ini file is picked up first though, might cause unintended issues elsewhere.